src/components/activity/ActivityDirectivesTable.svelte

```diff
@@ -214,9 +212,30 @@
       <ContextMenuItem on:click={scrollTimelineToActivityDirective}>Scroll to Activity</ContextMenuItem>
       <ContextMenuSeparator></ContextMenuSeparator>
     {/if}
-    {#if canPasteActivityDirectives()}
-      <ContextMenuItem on:click={pasteActivityDirectives}>{getPasteActivityDirectivesText()}</ContextMenuItem>
+    {#await getActivityDirectivesClipboardCount() then directivesInClipboard}
+      <ContextMenuItem
+        use={[
+          [
+            permissionHandler,
+            {
+              hasPermission: hasCreatePermission && directivesInClipboard > 0,
+              permissionError: () => {
+                if (planReadOnly) {
+                  return PlanStatusMessages.READ_ONLY;
+                } else if (!hasCreatePermission) {
+                  return 'You do not have permission create activity directives';
+                } else if (directivesInClipboard <= 0) {
+                  return 'No activity directives in clipboard';
+                }
```

Comment:
Can you add a final return statement for if the logic falls through all the conditionals?

Reply:
Do you mind saving the `permissionError` to a variable that gets updated in a reactive statement? We generally save the error to a local state variable so that it's calculated only when the dependent variables change and not every time the component re-renders.